src/libcore/fmt/mod.rs

```diff
@@ -1036,6 +1036,32 @@ pub fn write(output: &mut dyn Write, args: Arguments) -> Result {
     Ok(())
 }
 
+/// Padding after the end of something. Returned by `Formatter::padding`.
+#[must_use = "don't forget to write the post padding"]
+struct PostPadding {
+    fill: [u8; 4],
+    fill_len: u32,
+    padding: usize,
+}
+
+impl PostPadding {
+    /// Safety relies on `fill[..fill_len]` being a valid UTF-8 char.
+    unsafe fn new(fill: [u8; 4], fill_len: u32, padding: usize) -> PostPadding {
```

Comment:
Is it worth just having `fill` be `char` here and using `write_char` in `fn write` over `write_str`?

`write_char` is `write_str(c.encode_utf8(&mut [0; 4]))` in the general case. If I understand the code correctly we're calling encode_utf8 anyway to get `fill` right now so just passing the char in directly seems better.

Based on local testing it also seems to make the `PostPadding` struct get removed entirely (i.e., we're just passing around the underlying i32 and usize AFAICT).

Reply:
I was avoiding this since I thought the extra inlined copy of `encode_utf8` would add more code size, but it turns out that this somehow saves another 300 bytes!

Reply:
Another 150 bytes saved by using `write_char` in `fn padding` too.

Reply:
I presume that this is the case because the `fill` buffer here was already generated via `encode_utf8` anyway